processor/k8sprocessor/doc.go

```diff
@@ -15,13 +15,27 @@
 // Package k8sprocessor allow automatic tagging of spans, metrics and logs with k8s metadata.
 //
 // The processor automatically discovers k8s resources (pods), extracts metadata from them and adds the
-// extracted metadata to the relevant spans, metrics and logs. The processor use the kubernetes API to discover all pods
-// running in a cluster, keeps a record of their IP addresses and interesting metadata. Upon receiving telemetry data,
-// the processor looks for presence of well-known resource attributes which might contain IP address ("ip",
-// "k8s.pod.ip" for logs, metrics or traces and "host.name" for metrics). If this field is not available, or it
-// does not contain a valid IP address, the processor tries to identify the source IP address of the service
-// that sent the telemetry data.
-// If a match is found, the cached metadata is added to the data as resource attributes.
+// extracted metadata to the relevant spans, metrics and logs. The processor uses the kubernetes API to discover all pods
+// running in a cluster, keeps a record of their IP addresses, pod UIDs and interesting metadata.
```

Comment:
> and interesting metadata

I think I've seen only pod UID and IP addresses here. Am I missing something?

Reply:
UID and IP are used to associate metadata with spans metrics and logs. But this metadata contains other information like Pod Name, Deployment etc.

Reply:
Perhaps it's not part of this PR, but how would I use the pod name / deployment name and so on?